lib/cache/app.go

```diff
@@ -45,8 +47,13 @@ func newAppCollection(p services.Apps, w types.WatchKind) (*collection[types.App
 				appNameIndex: types.Application.GetName,
 			}),
 		fetcher: func(ctx context.Context, loadSecrets bool) ([]types.Application, error) {
-			apps, err := p.GetApps(ctx)
-			return apps, trace.Wrap(err)
+			out, err := stream.Collect(upstream.Apps(ctx, "", ""))
+			// TODO(tross): DELETE IN v21.0.0
+			if trace.IsNotImplemented(err) {
```

Comment:
Can we have the client's GetApps method use Apps with the fallback to the old rpc instead so we don't have to add a million fallbacks from Collect(Apps) to GetApps everywhere? There's a precedent for this, the client's GetNodes also does this.

Reply:
Should we deprecate the GetApps method then, at least in v19?

Reply:
Is there any reason to use the unpaginated endpoint? Even with the deprecation warning, I wonder about existing code using the api module, it would be nice if we could transparently make it not die if there's too many apps in a cluster (and I'm not even sure if we should deprecate methods in the middle of a major).

Reply:
Deprecate GetApps as a method and implement it using Collect+Resources+ListApps just for the transitional period, I guess?